### PR TITLE
Add support for Onesti EasyCode Touch lock

### PR DESCRIPTION
**Pull request by 1bard**
## Proposed change

This adds support for Onesti EasyCode Touch door lock.
The quirk removes incorrect MainsPowered-flag from NODE_DESCRIPTOR, making battery percentage visible in HA.

## Checklist

- [x] The changes are tested and work correctly
- [x] `pre-commit` checks pass / the code has been formatted using Black
- [ ] Tests have been added to verify that the new code works

### CODE REVIEW

**Comment by 1bard** [comment]
I have not checked "Tests have been added to verify that the new code works".
I really don't know what tests to add (or how).
I'm new to Python-development and there's not much code to test.

File: zhaquirks/onesti/easycodetouch.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+MANUFACTURER_SPECIFIC_PROFILE_ID = 0xFEA2
+
+
+class BatteryBasicCluster(CustomCluster, Basic):
+    """Basic cluster implementation.
+    This implementation sets power source to battery
+    and removes incorrect MainsPowered-flag.
+    """
+
+    cluster_id = Basic.cluster_id
```
Comment (TheJulianJES) [inline zhaquirks/onesti/easycodetouch.py:33]: This line shouldn't be needed, as inheriting `Basic` also sets that `cluster_id`.
(But I know it's also done in some other places still. You can just remove it though)

File: zhaquirks/onesti/easycodetouch.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    POWER_SOURCE = 0x0007
+
+    _CONSTANT_ATTRIBUTES = {
+        POWER_SOURCE: Basic.PowerSource.Battery,
```
Comment (TheJulianJES) [inline zhaquirks/onesti/easycodetouch.py:37]: Just curious, is this "power source" attribute actually used anywhere (in HA)?
Showing the battery percentage sensor only relies on the node descriptor IIRC(?)

File: zhaquirks/onesti/easycodetouch.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+            aps_flags=0,
+            frequency_band=8,
+            mac_capability_flags=136,
+            manufacturer_code=4660,
+            maximum_buffer_size=108,
+            maximum_incoming_transfer_size=127,
+            server_mask=11264,
+            maximum_outgoing_transfer_size=127,
+            descriptor_capability_field=0,
+        ),
```
Comment (TheJulianJES) [inline zhaquirks/onesti/easycodetouch.py:82]: "To compare": can you also post the original/unquirked device signature of the device?

File: zhaquirks/onesti/easycodetouch.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    cluster_id = Basic.cluster_id
+    POWER_SOURCE = 0x0007
+
+    _CONSTANT_ATTRIBUTES = {
+        POWER_SOURCE: Basic.PowerSource.Battery,
+    }
+
+
+class EasyCodeTouch(CustomDevice):
+    """onesti.easycodetouch"""
```
Comment (TheJulianJES) [inline zhaquirks/onesti/easycodetouch.py:42]:
```suggestion
    """Onesti EasyCodeTouch quirk."""
```

File: zhaquirks/onesti/easycodetouch.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+
+class BatteryBasicCluster(CustomCluster, Basic):
+    """Basic cluster implementation.
+    This implementation sets power source to battery
+    and removes incorrect MainsPowered-flag.
+    """
+
+    cluster_id = Basic.cluster_id
+    POWER_SOURCE = 0x0007
```
Comment (TheJulianJES) [inline zhaquirks/onesti/easycodetouch.py:34]: Getting attribute ids will be easier in the future, but right now, we can already do this:
```suggestion
    Basic.attributes_by_name["power_source"].id
```

**Review by TheJulianJES — COMMENTED** [review]
Thanks for the PR! Some small comments/questions:

File: zhaquirks/onesti/easycodetouch.py (hunk 2 above)
Comment (1bard) [reply zhaquirks/onesti/easycodetouch.py]: This was my initial attempt to make it act as a battery-powered device.
I've tested and you're right. Power source attribute is not required. I've removed it.

File: zhaquirks/onesti/easycodetouch.py (hunk 3 above)
Comment (1bard) [reply zhaquirks/onesti/easycodetouch.py:82]: Added

**Pull request closed without merging**